lightning/src/ln/chanmon_update_fail_tests.rs

```diff
@@ -3860,3 +3860,145 @@ fn test_claim_to_closed_channel_blocks_claimed_event() {
 	nodes[1].chain_monitor.complete_sole_pending_chan_update(&chan_a.2);
 	expect_payment_claimed!(nodes[1], payment_hash, 1_000_000);
 }
+
+#[test]
+#[cfg(feature = "std")]
+fn test_single_channel_multiple_mpp() {
+	// Test what happens when we attempt to claim an MPP with many parts that came to us through
+	// the same channel with a synchronous persistence interface which has very high latency.
+	//
+	// Previously, if a `revoke_and_ack` came in while we were still running in
+	// `ChannelManager::claim_payment` we'd end up hanging waiting to apply a
+	// `ChannelMonitorUpdate` until after it completed. See the commit which introduced this test
+	// for more info.
+	let chanmon_cfgs = create_chanmon_cfgs(7);
+	let node_cfgs = create_node_cfgs(7, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(7, &node_cfgs, &[None; 7]);
+	let mut nodes = create_network(7, &node_cfgs, &node_chanmgrs);
+
+	let node_5_id = nodes[5].node.get_our_node_id();
+	let node_6_id = nodes[6].node.get_our_node_id();
+
+	// Send an MPP payment in four parts along the path shown from top to bottom
+	//      0
+	//   1 2 3 4
+	//      5
+	//      6
+
+	create_announced_chan_between_nodes_with_value(&nodes, 0, 1, 100_000, 0);
+	create_announced_chan_between_nodes_with_value(&nodes, 0, 2, 100_000, 0);
+	create_announced_chan_between_nodes_with_value(&nodes, 0, 3, 100_000, 0);
+	create_announced_chan_between_nodes_with_value(&nodes, 0, 4, 100_000, 0);
+	create_announced_chan_between_nodes_with_value(&nodes, 1, 5, 100_000, 0);
+	create_announced_chan_between_nodes_with_value(&nodes, 2, 5, 100_000, 0);
+	create_announced_chan_between_nodes_with_value(&nodes, 3, 5, 100_000, 0);
+	create_announced_chan_between_nodes_with_value(&nodes, 4, 5, 100_000, 0);
+	create_announced_chan_between_nodes_with_value(&nodes, 5, 6, 1_000_000, 0);
+
+	let (mut route, payment_hash, payment_preimage, payment_secret) = get_route_and_payment_hash!(&nodes[0], nodes[6], 30_000_000);
+
+	send_along_route_with_secret(&nodes[0], route, &[&[&nodes[1], &nodes[5], &nodes[6]], &[&nodes[2], &nodes[5], &nodes[6]], &[&nodes[3], &nodes[5], &nodes[6]], &[&nodes[4], &nodes[5], &nodes[6]]], 30_000_000, payment_hash, payment_secret);
+
+	let (do_a_write, blocker) = std::sync::mpsc::sync_channel(0);
+	*nodes[6].chain_monitor.write_blocker.lock().unwrap() = Some(blocker);
+
+	// Until we have std::thread::scoped we have to unsafe { turn off the borrow checker }.
+	let claim_node: &'static TestChannelManager<'static, 'static> =
+		unsafe { std::mem::transmute(nodes[6].node as &TestChannelManager) };
```

Comment:
Looks like this is still the best way without scoped threads. Seems obvious from the `'static` lifetimes that this extends the lifetime, but maybe a link to the [example](https://doc.rust-lang.org/std/mem/fn.transmute.html#examples) may help readers for context.